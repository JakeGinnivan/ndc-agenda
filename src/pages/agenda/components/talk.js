// @flow
import React from 'react'
import { connect } from 'react-redux'
import { addToSchedule } from '../../../app.redux'

const AddToSchedule = (props) => (
    <button
        onClick={props.addToSchedule}
    >
        Add to schedule
    </button>
)

const RemoveFromSchedule = ({ removeFromSchedule} ) => (
    <button
        onClick={removeFromSchedule}
    >
        Remove from schedule
    </button>
)

export const TalkRaw = ({ startTime, title, link, dispatch, schedule }) => {
    const scheduleButton = schedule.indexOf(link) === -1
        ? <AddToSchedule addToSchedule={() => dispatch(addToSchedule(link))} />
        : <RemoveFromSchedule removeFromSchedule={() => {}} />
    return (
        <div>
            {startTime.hour}:{startTime.minutes} {title}
            {scheduleButton}
        </div>
    )
}

const mapStateToProps = (state) => {
    return { schedule: state.schedule }
}
export const Talk = connect(mapStateToProps)(TalkRaw)