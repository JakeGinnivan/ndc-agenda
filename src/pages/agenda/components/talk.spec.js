import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { TalkRaw } from './talk'

it('renders a talk correctly', () => {
    const dispatch = () => { }
    const talk = renderer.create(<TalkRaw
        startTime={{ hour: 9, minutes: 0}}
        title="Some Talk"
        link="http://ndcsydney.com/sometalk"
        dispatch={dispatch}
        schedule={[]} />)

    const rendered = talk.toJSON()

    expect(rendered).toMatchSnapshot()
})

it('another test', () => {
    const dispatched = []
    const dispatch = (action) => {
        dispatched.push(action)
    }
    const talk = mount(<TalkRaw
        startTime={{ hour: 9, minutes: 0}}
        title="Some Talk"
        link="http://ndcsydney.com/sometalk"
        dispatch={dispatch}
        schedule={[]} />)

    const rendered = talk.find('button')
    rendered.simulate('click', { button: 0 })

    expect(dispatched).toMatchSnapshot()
})