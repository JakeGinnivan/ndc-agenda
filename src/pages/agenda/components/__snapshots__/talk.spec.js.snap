// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`another test 1`] = `
<div>
  Some Talk
  <button
    onClick={[Function]}
  >
    Add to schedule
  </button>
</div>
`;

exports[`renders a talk correctly 1`] = `
<div>
  Some Talk
  <button
    onClick={[Function]}
  >
    Add to schedule
  </button>
</div>
`;
